### PR TITLE
add backend express server

### PR DESCRIPTION
**Pull request by EngrEric**
created a  backend for the Agroeasy client app.
Moved all the previous root contents of the app to client directory/folder.
created an express server at the root.

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    "reactstrap": "^6.1.0"
+  },
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test --env=jsdom",
+    "eject": "react-scripts eject"
+  },
+  "proxy":"http://localhost:5000"
+}
```
Comment (Way2nnadi) [inline client/package.json]: move this package.json to the root folder

Comment (Way2nnadi) [inline client/package.json]: combine this `package.json` with the one in the root folder

File: server.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+const express = require('express');
+const path = require('path');
+
+const app = express();
+const port = process.env.PORT || 5000;
+
+//API calls
+app.get('/api',(req,res)=>{
+    res.send({express:"Express at work here"});
+});
```
Comment (Way2nnadi) [inline server.js]: Remove lines `7-10` for now given that we do not have any API calls

File: server.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+if(process.env.NODE_ENV === 'production'){
+    //serve the static files
+    app.use(express.static(path.join(__dirname, 'client/build')));
+
+    //Handle react routing, return all requests to react app
+    app.get('*', function(req, res){
+        res.sendFile(path.join(__dirname, 'client/build', 'index.html'));
+    });
+}
```
Comment (Way2nnadi) [inline server.js]: Lines `14-19` should not be wrapped in an if statement for now. We will optimize the code later for `production` and `development` environments

File: client/package.json (hunk 1 above)
Comment (EngrEric) [reply client/package.json]: Is it only this part selected or the entire package.json of contents?

Comment (Way2nnadi) [reply client/package.json]: there should only be one `package.json` defined per project. Merge the contents of this one with the one in the root folder

**Pull request merged**